docs/design/coreclr/jit/jit-disasm.md

```diff
@@ -0,0 +1,71 @@
+# JIT disassembly
+
+The JIT contains several configuration parameters that cause it to output textual disassembly of the generated code.
+Many of these parameters are available in the shipping product (a.k.a. the "Release" build flavor), as well as the internal
+Debug/Checked build flavors.
+
+These parameters include:
+* `DOTNET_JitDisasmSumary` - set to 1 to print a list of all JIT compiled functions.
+* `DOTNET_JitDisasm` - output disassembly for the specified functions. E.g., `DOTNET_JitDisasm=Main`, `DOTNET_JitDisasm=Main Test1 Test2`, `DOTNET_JitDisasm=*` (for all functions).
+* `DOTNET_JitDisasmDiffable` - set to 1 to make the generated code "diff-able", namely, replace pointer values in the output with
+the same well-known, identical values, so they textually compare identically.
+* `DOTNET_JitDisasmWithAlignmentBoundaries` - set to 1 to display alignment boundaries in the generated code.
+* `DOTNET_JitDisasmWithCodeBytes` - set to 1 to display the actual code bytes in addition to textual disassembly.
+* `DOTNET_JitDisasmWithGC` - Debug/Checked builds only. Set to 1 to display GC information interleaved with the textual disassembly.
+* `DOTNET_JitDisasmWithDebugInfo` - Debug/Checked builds only. Set to 1 to display debug information (variable live ranges)
+interleaved with the textual disassembly.
+* `DOTNET_JitStdOutFile` - if not set, all JIT output goes to standard output. If set, it is the name of a file to which JIT output
+will be written.
+
+# The JIT late disassembler
+
+The disassembly displayed by `DOTNET_JitDisasm` is printed based on a JIT internal representation. An additional disassembler, called
+the "late disassembler", is available that disassembles the final code bytes. The late disassembler uses the coredistools package
+(based on LLVM) to interpret the code bytes. In this way, it provides a way to verify that the JIT disassembly matches a third-party
```

Comment:
> (based on LLVM)

might be good to point out the version of LLVM as well as coredistools package. I can understand that it can get outdated, but will be good to have and we will try to make sure it is up-to-date.